CRM/Core/BAO/Block.php

```diff
@@ -345,12 +346,14 @@ public static function create($blockName, &$params, $entity = NULL, $contactId =
 
       $blockFields = array_merge($value, $contactFields);
       if ($name === 'Email') {
-        // @todo ideally all would call the api which is our main tested function,
-        // and towards that call the create rather than add which is preferred by the
-        // api. In order to be careful with change only email is swapped over here because it
-        // is specifically tested in testImportParserWithUpdateWithContactID
-        // and the primary handling is otherwise bypassed on importing an email update.
-        $blocks[] = CRM_Core_BAO_Email::create($blockFields);
+        $emails = civicrm_api3('Email', 'create', $blockFields)['values'];
```

Comment:
@eileenmcnaughton Test failures relate, looks like `location_type_id` needs casting to int from bool, i.e. something silly like:

```php
if (!empty($blockFields['location_type_id'])) {
  $blockFields['location_type_id'] = (int) $blockFields['location_type_id'];
}
```

Reply:
@pfigel yeah - it all looks pretty alarming to be honest. I'm not sure this isn't surfacing a bug - will dig a little more